### DISCUSSION TITLE
net::ERR_HTTP2_PROTOCOL_ERROR

### DISCUSSION THREAD

**Discussion by Unfield** [pocketbase/pocketbase#1412] [Q&A]
Hey,
I use PocketBase Realtime in my Project. After a while, I get 'Failed to load resource: net::ERR_HTTP2_PROTOCOL_ERROR' printed in the Browser Console (coming from /api/realtime). What am I missing?

**Comment by ganigeorgiev** [comment]
It is hard to tell what is wrong without the actual network error response or code sample what you are doing.

Are you behind a reverse proxy (eg. nginx)?
Is the error reproduced locally?
When exactly do you get the error? By default the realtime connection will be autodisconnected if there were no new messages within a 5mins interval (the SDKs will handle the reconnect for you).

**Reply by luongngocminh** [reply]
I got this exact error recently running pocketbase behind a nginx proxy (to be more specific, behind Caprover PaaS). The error happens around 1min after connected. Running Pocketbase locally is fine.

**Reply by ganigeorgiev** [reply]
@luongngocminh Make sure that you've properly configured the read and write timeout of your nginx proxy. For an example minimal nginx config you can refer to https://pocketbase.io/docs/going-to-production/#using-reverse-proxy.

Note that realtime connection errors usually can be ignored because as mentioned above the SDKs will handle the reconnect for you.

**Reply by ganigeorgiev** [reply]
@luongngocminh I also forgot to mention that some VPS providers also sometimes have hard limits to how long a connection can remains open (the free plan in fly.io for example is ~60sec the last time I checked) so even if you have setup a higher limit in your proxy you can still stumble on the above error. But again, realtime connection errors are handled gracefully and the SDKs will try to reconnect whenever an issue occur (eg. the server went down, temp internet connection issues, etc.).

**Reply by luongngocminh** [reply]
@ganigeorgiev I've move pocketbase out off the Caprover PaaS with appropriate proxy config using your link and everything working good now.
